l1-contracts/contracts/bridgehub/Bridgehub.sol

```diff
@@ -163,6 +163,14 @@ contract Bridgehub is IBridgehub, ReentrancyGuard, Ownable2StepUpgradeable, Paus
         messageRoot = _messageRoot;
     }
 
+    /// @notice Used for the upgrade to set the baseTokenAssetId previously stored as baseToken.
+    /// @param _chainId the chainId of the chain.
+    function setLegacyBaseTokenAssetId(uint256 _chainId) external {
+        address token = baseToken[_chainId];
+        require(token != address(0), "BH: token not set");
+        baseTokenAssetId[_chainId] = DataEncoding.encodeNTVAssetId(block.chainid, token);
```

Comment:
should be checked that the chain does not yet have the base token asset id set.

Probably reverting in case the token is set is too much (since the method is public and be overrun), but at least not overwriting the base token should be done in case some chains have custom asset handler for base token